### PR TITLE
Feature: Get Destination Chain Transaction

### PR DESCRIPTION
**Pull request by napalmpapalam**
This pull request implements an endpoint to return withdrawal transaction information using the deposit transaction hash.  Such functionality is acquired with the "Bridge indexers", goroutines, that listen to withdrawal events emitted by the bridge contracts and store them in the database.

### Added
- Full config examples
- GET withdrawal by the deposit hash SSE endpoint
- `withdrawals` table to the database migrations
- Withdrawals cache querier

### Changed
- Refactored parsing of the genesis file for the collections and items indexers to omit race conditions and 
inconsistencies in the database
- Bridge Indexer to index the withdrawals events for the all chain types
- `txbuilder` package refactored to be able to create few transaction builders for the chains with the same type  

### Fixed
- Using UTC time in the all places
- Using one postgres connection for the all gorutines which could lock each other during execution database transactions
- `ChainTx` filter in the transfers select which could lead to the validation errors if the transaction hash is not in the
hex-encoded format
- `limit` filter in the transfers select which could lead to the empty response if the limit is not set

### CODE REVIEW

File: internal/services/bridge_producer/producers/evm/erc1155_handler.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	}
+
+	for iter.Next() {
+		if err = ctx.Err(); err != nil {
+			return errors.Wrap(err, "died by context")
+		}
+		e := iter.Event
+
+		if e == nil {
+			h.log.Error("got nil event")
```
Comment (ihordiachenko) [inline internal/services/bridge_producer/producers/evm/erc1155_handler.go:72]: Add some more context to the error if possible

**Review by ihordiachenko — CHANGES_REQUESTED** [review]

File: pkg/txbuild/builder.go [hunk 2]
```diff
@@ -19,25 +25,80 @@ var (
...
+		}
+
+		switch chain.Type {
+		case tokenmanager.NetworkType_EVM:
+			cli, err := ethclient.Dial(chain.Rpc)
+			if err != nil {
+				panic(errors.Wrap(err, "failed to dial eth client"))
+			}
+
+			builders[chain.Name] = ethtx.NewBuilder(cli, abi, common.HexToAddress(chainConf.contract))
```
Comment (ihordiachenko) [inline pkg/txbuild/builder.go:76]: Are these builder structs immutable? Please make sure there are no race conditions when working with this `builders` map when using a `MultiBuilder`

File: pkg/txbuild/builder.go [hunk 3]
```diff
@@ -19,25 +25,80 @@ var (
...
+				solana.PublicKeyFromBytes(hexutil.MustDecode(chainConf.contract)),
+				solana.PublicKeyFromBytes(hexutil.MustDecode(chainConf.adminPublicKey)),
+			)
+		case tokenmanager.NetworkType_Near:
+			cli, err := nearclient.New(chain.Rpc)
+			if err != nil {
+				panic(errors.Wrap(err, "failed to dial near client"))
+			}
+			builders[chain.Name] = neartx.NewBuilder(cli, string(hexutil.MustDecode(chainConf.contract)))
+		case tokenmanager.NetworkType_Other: // FIXME: change to the rarimo chain type when it will be implemented
```
[the remark this answers is not in the document]
Comment (napalmpapalam) [reply pkg/txbuild/builder.go:93]: not yet

File: pkg/txbuild/builder.go (hunk 2 above)
Comment (napalmpapalam) [reply pkg/txbuild/builder.go:76]: MultiBuilder actually creates just once during injecting it into the api router middleware

**Review by ihordiachenko — APPROVED** [review]

**Pull request merged**